src/libostree/ostree-repo-commit.c

```diff
@@ -308,16 +308,19 @@ commit_loose_object_trusted (OstreeRepo        *self,
 
 typedef struct
 {
-  gsize unpacked;
-  gsize archived;
+  OstreeObjectType objtype;
+  guint64 unpacked;
+  guint64 archived;
```

Comment:
Not changing these was definitely sloppy, but unfortunately due to the return type of `g_output_stream_splice` we still can't handle files with a compressed size that doesn't fit in a signed 32 bit integer. I'll change everything that I can, which will make at least the uncompressed sizes always right, but what can we do about that other case?